### ISSUE TITLE
fmriprep gets stuck on cluster

### ISSUE THREAD

**Issue by divyanganar** [poldracklab/fmriprep#1478]
Hi everyone! 

I'm super new to neuroimaging and just learning the ropes at the moment. The version of fmriprep-docker I'm using is 1.2.6 but the same problem did arise with 1.2.5.

So I'm also trying to run a few subjects on the spartan cluster (University of Melb). 

It goes up to a certain point and just hangs -- it doesn't throw an error, neither does it proceed beyond that point. I've tried several times and it has stayed at that point for >14 hours until the cluster cancels my job because of the walltime.

Script:

```#!/bin/bash
#SBATCH --nodes=1
#SBATCH --ntasks=1
#SBATCH --cpus-per-task=8
#SBATCH --time=0-30:00:00
#SBATCH --job-name=fmriprep_test
#SBATCH --mem=32G

myproject=$'/data/fmriprep'

module load Singularity/2.5.0-GCC-6.2.0

cd $myproject/fmriprep/t3_fmri_test_17212019

singularity run --cleanenv "${myproject}/fmriprep_image/fmriprep-1.2.5.simg" \
"${myproject}/t3_fmri_test_17212019/fmriprep_bids_input_data/" "${myproject}/fmriprep_output/" \
participant \
--fs-license-file "${myproject}/t3_fmri_test_17212019/license.txt" --use-aroma
```

It runs for around 10-11 hours and then reaches the point at which it stops running.

The last two lines before it stops are these:

```[Node] Running "lta_2_itk" ("niworkflows.interfaces.freesurfer.PatchedLTAConvert"), a CommandLine Interface with command: lta_convert --inlta /data/cephfs/punim0752/fmriprep/orwa_1sub_8cpus/work/fmriprep_wf/single_subject_01_wf/anat_preproc_wf/surface_recon_wf/t1_2_fsnative_xfm/out.lta --outitk /data/cephfs/punim0752/fmriprep/orwa_1sub_8cpus/work/fmriprep_wf/single_subject_01_wf/anat_prep oc_wf/anat_derivatives_wf/lta_2_itk/out.txt 190121-12:45:17,269 nipype.workflow INFO:```
	``` [Node] Finished "fmriprep_wf.single_subject_01_wf.anat_preproc_wf.anat_derivatives_wf.lta_2_itk".```

Output file:

[spartan_log.txt](https://github.com/poldracklab/fmriprep/files/2781561/spartan_log.txt)

This happened to me multiple times (with different subjects) on the cluster and I'm trying to figure out if I'm doing something wrong or if there's anything I need to change.

Any help would be greatly appreciated! Thank you in advance!

**Comment by divyanganar** [comment]
So I ran a subject last night with the verbose argument. Seems to have been stuck for quite a long time also. Attaching the output file from that:

[job_7262726_1.txt](https://github.com/poldracklab/fmriprep/files/2784958/job_7262726_1.txt)

**Comment by effigies** [comment]
I didn't see any errors in the logs. How are you verifying that it's no longer running? I see that `autorecon2_vol` starts, but does not finish. Perhaps it was still working on the FreeSurfer portions of the workflow?

**Comment by divyanganar** [comment]
Hi, yes sorry, there have been no errors. I am just concerned about the amount of time it seems to be taking? How long should it typically take with 8 cpus and 32G memory? I've let it run for as long as 35 hours (different instance) and it didn't move from that step -- was wondering if that's normal? I've just rerun the subject after it hit the wall time using --debug -vvvv and the same working and output directories. 

It's been at this for a while:

```
190122-12:42:33,944 nipype.workflow INFO:
	 [MultiProc] Running 1 tasks, and 0 jobs ready. Free memory (GB): 51.62/56.62, Free processors: 1/8.
                     Currently running:
                       *fmriprep_wf.single_subject_01_wf.anat_preproc_wf.surface_recon_wf.autorecon_resume_wf.autorecon2_vol
```
[job_7263009_1.txt](https://github.com/poldracklab/fmriprep/files/2785174/job_7263009_1.txt)

Maybe I should run a bunch of subjects and give a wall time of 100 hours and see?

**Comment by divyanganar** [comment]
Here's the log for a fresh run I did last night as well (just in case it helps). It was printing "cannot allocate job for ~11 hours and then resumed autorecon1 -- is that because of resources?

[job_7262726_1.txt](https://github.com/poldracklab/fmriprep/files/2785183/job_7262726_1.txt)

**Comment by divyanganar** [comment]
I ran the subject (same as my fist comment) for 35 more hours (~50 in total) and it still didn't finish. Here's the output file, it appears to not be able to allocate a particular job and get stuck.

[job_7263009_1.txt](https://github.com/poldracklab/fmriprep/files/2790430/job_7263009_1.txt)

**Comment by effigies** [comment]
The issue seems to be job `_autorecon_surfs0` running for over 21 hours.

```
	 [Node] Running "_autorecon_surfs0" ("nipype.interfaces.freesurfer.preprocess.ReconAll"), a CommandLine Interface with command:
recon-all -autorecon-hemi lh -noparcstats -nocortparc2 -noparcstats2 -nocortparc3 -noparcstats3 -nopctsurfcon -nohyporelabel -noaparc2aseg -noapas2aseg -nosegstats -nowmparc -nobalabels -openmp 7 -subjid sub-01 -sd /data/cephfs/punim0752/fmriprep/dicom_1005034/dicom_1005034_fmriprep_output/freesurfer 
190123-02:20:15,166 nipype.interface INFO:
	 resume recon-all : recon-all -autorecon-hemi lh -noparcstats -nocortparc2 -noparcstats2 -nocortparc3 -noparcstats3 -nopctsurfcon -nohyporelabel -noaparc2aseg -noapas2aseg -nosegstats -nowmparc -nobalabels -openmp 7 -subjid sub-01 -sd /data/cephfs/punim0752/fmriprep/dicom_1005034/dicom_1005034_fmriprep_output/freesurfer 
190123-02:20:16,177 nipype.workflow DEBUG:
	 Progress: 250 jobs, 75/1/1 (done/running/ready), 1/174 (pending_tasks/waiting).
190123-02:20:16,178 nipype.workflow DEBUG:
	 Tasks currently running: 1. Pending: 1.
190123-02:20:16,179 nipype.workflow INFO:
	 [MultiProc] Running 1 tasks, and 1 jobs ready. Free memory (GB): 51.62/56.62, Free processors: 1/8.
                     Currently running:
                       * _autorecon_surfs0
190123-02:20:16,270 nipype.workflow DEBUG:
	 Cannot allocate job 249 (5.00GB, 7 threads).
...
190123-23:40:48,37 nipype.workflow DEBUG:
	 Cannot allocate job 249 (5.00GB, 7 threads).
slurmstepd: error: *** JOB 7263009 ON spartan-rc170 CANCELLED AT 2019-01-24T10:40:48 DUE TO TIME LIMIT ***
```

You can check the logs in`freesurfer/sub-01/scripts/` to see what they say. There should be one that has `lh` in the name. As an initial guess, I'd suppose that the topological correction is having a lot of trouble, as that's the slowest part of the FreeSurfer pipeline. Have you looked at your T1w images and made sure that they don't have artifacts? If so, could you overlay your brain mask (both of these should be in your outputs folder by this point) and make sure that looks sensible?

**Comment by divyanganar** [comment]
We've actually used this data before for other analyses (a few years ago with SPM) and done the freesurfer recon on them (with an older version) and there were no problems then. 

Given these issues, I'm considering running freesurfer separately and then running fmriprep after putting the freesurfer outputs in (have tried it on the local machine and it worked) -- how many cores and memory would you recommend if freesurfer is already complete?

**Comment by oesteban** [comment]
Hi @divyanganar, did you finally resolve this issue?

**Comment by divyanganar** [comment]
Hi @oesteban ! I ran freesurfer separately and ran fmriprep on the cluster and it finished for most subjects (except a couple for whom freesurfer didn't finish at all or failed). That seemed to resolve the issue. I tried adding more arguments (omp threads, nthreads, ncpus) but that didn't change anything. Not sure what was going wrong but the pipeline never finished for us if we hadn't already done freesurfer. Quite bizarre because freesurfer ran pretty smoothly otherwise.

**Comment by oesteban** [comment]
Glad you finally got this to run. I'll let @effigies close this one if nothing jumps out that we should take care of.

**Comment by effigies** [comment]
Hi, sorry I dropped the ball on this. Yeah, it's very strange that you were able to run FreeSurfer independently, but not via fMRIPrep. As I alluded to in my last post, the only reasonable explanation I can think of involves the T1w brain mask produced by fMRIPrep. So I would keep an eye out to make sure those look sensible.

Otherwise, glad you were able to run it, in the end.

**Comment by divyanganar** [comment]
No worries, just glad we got it working. 
Actually, what's even weirder is that even the failure wasn't reliable. We've managed to run it on a few subjects (with freesurfer as part of the fmriprep pipeline) and it finished but most of the time it wouldn't, that's why we decided that it was best to do freesurfer separately. I'm thinking its something to do with our specific cluster? Maybe it doesn't allow certain configurations or something like that which prevents fmriprep from running smoothly?
